packages/data/src/registry.js

```diff
@@ -377,6 +326,21 @@ export function createRegistry( storeConfigs = {} ) {
 		subscribe,
 		select,
 		dispatch,
-		setupPersistence,
+		namespaces,
+		use,
```

Comment:
So basically, the API changes are:

 - making the `namespaces` array a public API.

I think it's fine.

The issue we'll have is how to update the default registry since it's not mutable. The `use` generates a new registry? Should we have a "special" `use` function for the default registry?

Reply:
It doesn't so much create a new registry as much as its own internal representation. `withPlugins` effectively makes what is exposed behave as a proxy to the current internal representation. It was partly done this way so that plugins act as layers, where the `registry` it receives when called is either the original implementation, or whatever was last overridden by a plugin.

Reply:
>making the namespaces array a public API.

Yes, I wasn't so sure about this. I still think it could be possible to omit, or limit (to e.g. an array of the registered `reducerKey`).

Since `registerStore` returns the actual store instance, at the worst case the plugin overridden value could inject the `triggerPersist` as part of the store's own `dispatch` / `getState`.

Reply:
I missed `withPlugins`, clever :)

Reply:
Sometimes I wonder if we should "deprecated" the use of the global registry entirely but that's a different subject.

Reply:
I think it's a nice convenience to have. As made evidenced here, it does make some things more difficult, like the idea of a "global plugin", which must be injected before anything else creates a store of its own (the `wp_add_inline_script` on "NUX" is not super obvious or easy to maintain).